osu.Game/Screens/Play/HUD/SpectatorList.cs

```diff
@@ -73,6 +73,9 @@ private void load(OsuColour colours)
             };
 
             HeaderColour.Value = Header.Colour;
+
+            ((IBindableList<SpectatorUser>)Spectators).BindTo(client.WatchingUsers);
+            ((IBindable<LocalUserPlayingState>)UserPlayingState).BindTo(gameplayState.PlayingState);
```

Comment:
Because this is a threaded component, these should not be bound on the BDL thread. Especially and _in particular_ the `BindableList<>`.

There is a potential race condition where:
- This thread is in the middle of copying elements from the remote list on the BDL thread.
- The remote list changes on the Update thread.

I believe these days we use `LoadComplete()` to bind bindables in general.